test/suite/wtbackup.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2021 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+import os
+import wttest
+from suite_subprocess import suite_subprocess
+from helper import compare_files
+
+# Shared base class used by backup tests.
+class backup_base(wttest.WiredTigerTestCase, suite_subprocess):
+    #
+    # Add data to the given uri.
+    # options:
+    #   cursor_options: a config string for cursors
+    #   num_ops: number of operations added to uri
+    #   mult: counter to have variance in data
+    #   counter: a counter to used to produce unique backup ids
+    #   session_checkpoint: boolean in which enables session checkpoints after adding data
+    #   initial_backup: To determine whether to increase/decrease counter, which determines
+    #           the backup id, and is generally used only for using add_data() first time.
+    #
+    def add_data(self, uri, key, val, options):
+        c = self.session.open_cursor(uri, None, options.get('cursor_options'))
```

Comment:
I think using an options "dictionary" to pass around values, feels a bit cumbersome.  Rather, I would have the shared base class create the variables with their defaults:
```
class backup_base(....):
   nops = 100
   mult = 0
   # etc....
```
And the other classes can override them:
```
class test_backup16(backup_base):
   mult=1
   nops=10
   # etc....
```
Or the scenarios can override the values if needed.  (I'm not sure we do that now, but we could have a scenario with `nops=1000` in the dict.  Then, we should be able to use `self.counter`, `self.mult`, `self.nops` directly.  For example:
```
self.data_options['counter'] += 1
```
can be like the original:
```
self.counter += 1
```
I think that will be easier to understand and maintain.  And the number of changes needed to be made should be reduced.

Reply:
sounds good, i've made the changes, only backup17 required an special change, let me know how you think of that test :D